### PR TITLE
Added a debug static variable to the interpreter

### PR DESCRIPTION
**Pull request by Strasser-Pablo**
Hi,

I added a debug static variable to print some debug information which were commented in the original code.
I'm not sure this code should stay in in release however I share it as it could be useful.

Have a nice day.

### CODE REVIEW

**Comment by jamesr66a** [comment]
Cool! Maybe we can also make this a gflag to make it runtime configurable. Thoughts, @zdevito @apaszke?

File: torch/csrc/jit/interpreter.cpp [hunk 1]
```diff
@@ -48,6 +48,8 @@ namespace torch { namespace jit {
 //   indicating whether this is the last use of the value. The interpreter
 //   should generate a move rather than a copy in this case.
 
+bool Code::debug=false;
```
Comment (zdevito) [inline torch/csrc/jit/interpreter.cpp]: Can you add a way to set this from python (see jit/init.cpp for where)?

File: torch/csrc/jit/interpreter.cpp [hunk 2]
```diff
@@ -659,17 +661,23 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
     size_t last = instructions.size();
 
     while (pc < last) {
-        // std::cout << "executing " << pc << ": ";
-        // function->dumpInstruction(std::cout, pc);
-        // std::cout << "\n";
+        if(Code::debug)
+        {
```
Comment (zdevito) [inline torch/csrc/jit/interpreter.cpp]: Can you fix the formatting to match the surround style?
```
if (Code::debug) {
```

Comment (Strasser-Pablo) [reply torch/csrc/jit/interpreter.cpp]: Ok I will do it in a few days when I have some time.

**Review by apaszke — COMMENTED** [review]
Did anyone ever use those statements for debugging after the interpreter was written? I haven’t, and I’m not sure if it’s worth adding this, since it adds some complexity to the code.

**Comment by apaszke** [comment]
Well the debug statements you've added are kind of low level at this point. If you simply want to make sure your graphs are correct, you're much better off doing `graph->lint()`.

**Comment by ezyang** [comment]
@Strasser-Pablo @apaszke What should we do with this PR?

**Comment by Strasser-Pablo** [comment]
For me you can close it given previous @apaszke comment and I understand that this modification add code complexity without much help for a lot of people except people needing low level access or writing directly in C++.

**Comment by apaszke** [comment]
I honestly can't remember a single time I needed those prints, so I'd vote to close this. Thanks @Strasser-Pablo!

**Pull request closed without merging**